spinn_front_end_common/interface/interface_functions/dsg_region_reloader.py

```diff
@@ -29,96 +28,149 @@
 from spinn_front_end_common.data import FecDataView
 from spinn_front_end_common.utilities.helpful_functions import (
     generate_unique_folder_name)
+from spinn_front_end_common.utilities.constants import BYTES_PER_WORD
 
 
-def dsg_region_reloader():
-    reloader = _DSGRegionReloader()
-    # pylint: disable=protected-access
-    reloader._run()
+def reload_dsg_regions():
+    """ Reloads DSG regions where needed
+    """
+    # build file paths for reloaded stuff
+    data_dir = get_reload_data_dir()
+
+    progress = ProgressBar(
+        FecDataView.get_n_placements(), "Reloading data")
+    for placement in progress.over(FecDataView.iterate_placemements()):
+        # Generate the data spec for the placement if needed
+        regenerate_data_spec(placement, data_dir)
+
+    # App data directory can be removed as should be empty
+    os.rmdir(data_dir)
 
 
-class _DSGRegionReloader(object):
-    """ Regenerates and reloads the data specifications.
+def get_reload_data_dir():
+    """ Get a path in which reloaded data files can be written
+
+    :rtype: str
+    """
+    run_dir_path = FecDataView.get_run_dir_path()
+    data_dir = generate_unique_folder_name(
+        run_dir_path, "reloaded_data_regions", "")
+    if not os.path.exists(data_dir):
+        os.makedirs(data_dir)
+    return data_dir
+
+
+def regenerate_data_spec(placement, data_dir):
+    """ Regenerate a data specification for a placement
+
+    :param ~.Placement placement: The placement to regenerate
+    :param str data_dir: A place to use to write data to
+    :return: Whether the data was regenerated or not
+    :rtype: bool
+    """
+    vertex = placement.vertex
+
+    # If the vertex doesn't regenerate, skip
+    if not isinstance(vertex, AbstractRewritesDataSpecification):
+        return False
```

Comment:
Why return a Value if none ever reads it

Reply:
Ah I should have been clear that I wanted to use this in a different branch; this means that the function can be used in an extension (which I am doing elsewhere).